addons/binding/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderLocalWatcherHandler.java

```diff
@@ -0,0 +1,234 @@
+     *
+     * triggerChannel(CHANNEL_LOCALFILENAME, newFtpFile);
+     *
+     * try {
+     * Thread.sleep(3000);
+     * } catch (InterruptedException e) {
+     *
+     * }
+     * }
+     * previousLocalListing = (ArrayList<String>) currentLocalListing.clone();
+     * }
+     * }
+     */
+
+    void listFiles(String path, boolean listHidden, boolean recursive) throws IOException {
+        File f = null;
+        File[] fileList;
+
+        try {
+
+            // create new file
+            f = new File(path);
+
+            // array of files and directory
+
+            fileList = f.listFiles(new FileFilter() {
+                @Override
+                public boolean accept(@Nullable File fileRaw) {
+                    if (listHidden) {
+                        return true;
+                    } else {
+                        return !fileRaw.isHidden();
+                    }
+                }
+            });
+
+            // for each name in the path array
+            if (fileList != null && fileList.length > 0) {
+                for (File file : fileList) {
+
+                    if (file.isDirectory()) {
+
+                        if (recursive) {
+                            listFiles(file.getAbsolutePath(), listHidden, recursive);
+                        }
+                    }
+                    if (file.isFile()) {
+                        // logger.debug(file.getAbsolutePath());
+                        currentLocalListing.add(file.getAbsolutePath());
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    protected Runnable periodicExecutionRunnable = new Runnable() {
+        @Override
+        public void run() {
+
+            String RootDir = config.localDir;
+            ArrayList<String> diffLocalListing = new ArrayList<String>();
+
+            /*
+             * if (ftpRootDir.endsWith("/")) {
+             * ftpRootDir = ftpRootDir.substring(0, ftpRootDir.length() - 1);
+             * }
+             *
+             * if (!ftpRootDir.startsWith("/")) {
+             * ftpRootDir = "/" + ftpRootDir;
+             * }
+             */
+            currentLocalListing.clear();
+
+            try {
+                listFiles(RootDir, config.listHiddenLocal, config.listRecursiveLocal);
+            } catch (IOException e) {
+
+                logger.debug("Can't read directory: {}", config.localDir);
+            }
+
+            try {
+                diffLocalListing = FileTools.listNewFiles(previousLocalListing, currentLocalListing,
+                        currentLocalListingFile);
+            } catch (IOException e1) {
+
+                logger.debug("Error geeting new file names");
+            }
+
+            for (String newLocalFile : diffLocalListing) {
+
+                triggerChannel(CHANNEL_LOCALFILENAME, newLocalFile);
+
+                try {
+                    Thread.sleep(3000);
```

Comment:
why the sleep?
If this is necessary at all it should be configurable

Reply:
Using this as a workaround. When I tested rule that would log file name when the new file is detected I faced a problem that if there are multiply new files the channel is triggered so fast that rule can only log last file name, but multiply times.

Reply:
How did your rule access the filename?
If it gets it from the trigger event directly it should work (I think)

Reply:
Can't tell, I'll test it again. The only thing that I remember that I was using sendTelegram every time the rule is triggering, maybe it was the issue. I'll re-tests without sleep timer again